src/sentry/static/sentry/app/views/settings/organizationMembers/organizationRequestsView.tsx

```diff
@@ -0,0 +1,157 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {AccessRequest, Member} from 'app/types';
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
+import {Panel, PanelBody, PanelHeader} from 'app/components/panels';
+import {t, tct} from 'app/locale';
+import AsyncView from 'app/views/asyncView';
+import EmptyMessage from 'app/views/settings/components/emptyMessage';
+import routeTitleGen from 'app/utils/routeTitle';
+import SentryTypes from 'app/sentryTypes';
+
+import InviteRequestRow from './inviteRequestRow';
+import OrganizationAccessRequests from './organizationAccessRequests';
+
+type Props = AsyncView['props'] & {
+  requestList: AccessRequest[];
+  inviteRequests: Member[];
+  updateInviteRequests: (id: string) => void;
+  updateRequestList: (id: string) => void;
+  showInviteRequests: boolean;
+};
+
+type State = AsyncView['state'] & {
+  inviteRequestBusy: Map<string, boolean>;
+};
+
+class OrganizationRequestsView extends AsyncView<Props, State> {
+  static propTypes = {
+    params: PropTypes.object,
+    requestList: PropTypes.array,
+    inviteRequests: PropTypes.array,
+    updateInviteRequests: PropTypes.func,
+    updateRequestList: PropTypes.func,
+    showInviteRequests: PropTypes.bool,
+  };
+
+  static contextTypes = {
+    router: PropTypes.object,
+    organization: SentryTypes.Organization,
+  };
+
+  static defaultProps = {
+    inviteRequests: [],
+  };
+
+  state: State = {
+    inviteRequestBusy: new Map(),
+    ...this.getDefaultState(),
+  };
+
+  getEndpoints() {
+    return [];
+  }
+
+  getTitle() {
+    const {orgId} = this.props.params;
+    return routeTitleGen(t('Requests'), orgId, false);
+  }
+
+  handleApprove = async (id: string, email: string) => {
+    const {
+      params: {orgId},
+      updateInviteRequests,
+    } = this.props;
+
+    this.setState(state => ({
+      inviteRequestBusy: state.inviteRequestBusy.set(id, true),
+    }));
+
+    try {
+      await this.api.requestPromise(`/organizations/${orgId}/invite-requests/${id}/`, {
+        method: 'PUT',
+        data: {approve: 1},
+      });
+      updateInviteRequests(id);
+      addSuccessMessage(tct('[email] has been invited', {email}));
+    } catch (err) {
+      addErrorMessage(tct('Error inviting [email]', {email}));
+      throw err;
+    }
+
+    this.setState(state => ({
+      inviteRequestBusy: state.inviteRequestBusy.set(id, false),
+    }));
+  };
+
+  handleDeny = async (id: string, email: string) => {
```

Comment:
If you wanted, you could generalize this function into a signel `handleAction({id, email, method, successMessage, errorMessage})`

and then have something like

```js
handleApprove = (id, email) =>
  this.handleAction({
    id, 
    email,
    method: 'PUT',
    successMessage: tct('[email] has been invited', {email}),
    errorMessage: tct('Error inviting [email]', {email}),
  });
```